python/activity_stream/widget_activity_stream_base.py

```diff
@@ -0,0 +1,220 @@
+    
+    @property
+    def entity_type(self):
+        return self._entity_type
+    
+    @property
+    def entity_id(self):
+        return self._entity_id
+    
+    @property
+    def activity_id(self):
+        return self._activity_id
+
+    def set_info(self, data):
+        """
+        Populate text fields for this widget
+        
+        :param data: data dictionary with activity stream info. 
+        """
+        self._target_entity_type = data["primary_entity"]["type"] 
+        self._target_entity_id = data["primary_entity"]["id"]
+        self._activity_id = data["id"]
+        
+    def set_thumbnail(self, data):
+        """
+        Populate the UI with the given thumbnail
+        """
+        # each deriving class should implement this
+        raise NotImplementedError
+    
+    ##############################################################################
+    # protected methods called by base classes
+        
+    def mouseDoubleClickEvent(self, event):
+        """
+        Forward any clicks from this widget
+        """
+        self.entity_requested.emit(self._target_entity_type, self._target_entity_id)
+            
+    def _icon_url_for_entity_type(self, entity_type):
+        """
+        Given an entity type, return a dark 16x16px icon resource url.
+        Returns None if no icon exists
+        
+        :param entity_type: entity type
+        :returns: resource url or None
+        """
+        entity_types_with_icons = ["Asset", 
+                                   "ClientUser",
+                                   "EventLogEntry",
+                                   "Group",
+                                   "HumanUser",
+                                   "Note",
+                                   "Playlist",
+                                   "Project",
+                                   "Sequence",
+                                   "Shot",
+                                   "Task",
+                                   "Ticket",
+                                   "Version",
+                                   ]
+
+        if entity_type in entity_types_with_icons:        
+            url = ":/tk_framework_qtwidgets.activity_stream/entity_icons/icon_%s_dark.png" % entity_type
+        else:
+            url = None
+            
+        return url            
+            
+    def _entity_request_from_url(self, url):
+        """
+        Helper method.
+        
+        Given a url on the form entity_type:entity_id, 
+        emit an entity_requested signal. This is typically
+        used by deriving classes to hook up linkActivated
+        signals to. 
+        """
+        try:
+            (entity_type, entity_id) = url.split(":")
+            entity_id = int(entity_id)
+            self.entity_requested.emit(entity_type, entity_id)
+        except:
+            self._bundle.log_warning("Could not parse url '%s'" % url)
+
+        
+    def _set_timestamp(self, data, label):
```

Comment:
this is currently not shared. My thinking was that we can potentially share it at the point of reuse.